### ISSUE TITLE
[REQUEST]: Consider not tying to the Ubunutu glibc

### ISSUE THREAD

**Issue by mattgodbolt** [compiler-explorer/compiler-explorer#3103]
### Is your feature request related to a problem? Please describe

A user emailed to ask what glibc we used, as they wanted to try out something in a more contemporary (~1 year old) glibc. We use the glibc that comes with the OS (currently 2.31)

### Describe the solution you'd like

I realised we don't have a good answer to "what if we wanted to use a newer glibc" short of upgrading the OS.

### Describe alternatives you've considered

Not sure what to do here, but thought I'd capture this for ideas.

### Additional context

Not applicable

**Comment by partouf** [comment]
I know the linking is "usually" done through the files:
```
/etc/ld.so.conf
/etc/ld.so.conf.d/*
```

But a compiler might also be using hardcoded paths.

The includes are a little bit more complicated probably, but basically under `/usr/include` is part of it.

So in theory we could "mount" different things in those places using nsjail?

**Comment by narpfel** [comment]
The glibc docs have a section on using a glibc installation in a nonstandard location: https://sourceware.org/glibc/wiki/Testing/Builds#Compile_against_glibc_in_an_installed_location

TLDR is AFAIU that passing `-I...`, `-L...`, `-Wl,-rpath=...` (which is already possible) and `-Wl,--dynamic-linker=...` should be sufficient.

Testing locally, `-Wl,--dynamic-linker` didn’t seem necessary, however my system-wide version of glibc is 2.43 and I tested with the current git `master` (between 2.43 and 2.44), so it’s possible that it might be different for a bigger version difference.
